### PR TITLE
Add access locks to gas canisters

### PR DESCRIPTION
**Pull request by JustinTrotter**
## About the PR 
During shifts cargo sometimes wants to sell Plasma/Tritium/Frezon filled gas canisters to make the big bucks; however, then as it's being transported to cargo, the clown slips the cargo-tech hauling it and as a joke opens the canister, que shuttle call. So to mitigate these unfortunate incidents, we can install access readers to all canisters, defaulting all to unlocked except for dangerous gases like Plasma, Frezon, Tritium, Miasma, and NitrousOxide.

Locking/Unlocking behaves the same as any locker, just alt + left click or use the content menu to unlock.
Access is granted to only Atmospherics and Research Director privileges. (I figure that when toxin bombs eventually become a thing, the RD would be needed to unlock the canisters for his research team in order to set up a chain of custody.)

Antags are still be able to use emags to unlock the canisters. Also shitters could still bash the thing open.

Cargo selling locked canisters has no effect on its value.

I've added two led indicators on each gas canister to signify its locked status at a glance. Red for locked, Green for unlocked.

Attempting to hand interact with a locked canister without access results in a pop up message "Access denied." with a standard denied beeping sound. Like with lockers, hand interacting with a locked canister with the needed credentials will automatically unlock it and open the interface.

**Screenshots**

Screenshot_2022-08-13_22-53-32

**Changelog**

:cl:
- add: Gas canisters now come with access locks.

### CODE REVIEW

**Comment by mirrorcult** [comment]
this isn't on any server i'm aware of, but im down to try it honestly

File: Content.Server/Atmos/Piping/Unary/EntitySystems/GasCanisterSystem.cs [hunk 1]
```diff
@@ -73,6 +81,9 @@ private void OnCanisterStartup(EntityUid uid, GasCanisterComponent canister, Com
             {
                 containerManager.MakeContainer<ContainerSlot>(canister.ContainerName);
             }
+
+            if (TryComp<LockComponent>(uid, out var lockComponent))
+                _appearanceSystem.SetData(uid, GasCanisterVisuals.Locked, lockComponent.Locked);
```
Comment (mirrorcult) [inline Content.Server/Atmos/Piping/Unary/EntitySystems/GasCanisterSystem.cs:86]: this should really be a generic thing that lockcomponent handles (and renamed to `LockVisuals` or something) but I won't enforce that it's done in this PR

**Review by mirrorcult — APPROVED** [review]

**Comment by Elijahrane** [comment]
Should every canister be locked? The O2 canisters are meant for people to refill their oxygen tanks

**Comment by HoidC** [comment]
> Should every canister be locked? The O2 canisters are meant for people to refill their oxygen tanks

"defaulting to all unlocked except dangerous canisters like Plasma, Frezon, Tritium, Miasma, and NitrousOxide.

**Comment by Elijahrane** [comment]
> if i'm reading this right o2 is unlocked by default

Should it have the lock component at all though

**Comment by Emisse** [comment]
it should be lockable

**Comment by HoidC** [comment]
> > if i'm reading this right o2 is unlocked by default
> 
> Should it have the lock component at all though

I mean, why not? If I want to move an o2 canister through a hallway (for whatever reason) and don't want the above thing happening I'd like to lock it so people don't open it.

The only people who can lock them aren't traitors, so I doubt griefing would happen.

**Comment by JustinTrotter** [comment]
> The only people who can lock them aren't traitors, so I doubt griefing would happen. 

Right this is why I decided not to include engineering on the access list. The only once that should have the power to grant or prevent access would be atmos and relevant heads like Cap, RD, and CE.

**Comment by Cheackraze** [comment]
what do you think about leaving nitrous oxide off the locked list. I think co2 is more dangerous than nitrous.

**Review by vulppine — APPROVED** [review]
personally see no issues with this as a gameplay feature as there's still a couple ways for a player who really wants to do bad things with gas canisters to get around this limitation, this is more towards preventing immediately taking a canister and spilling it everywhere

make sure you do a sync with the master branch though

**Review by mirrorcult — APPROVED** [review]

**Pull request merged**